### PR TITLE
use updated dcrlibwallet PurchaseTicket method

### PR DESCRIPTION
**Pull request by itswisdomagain**
Also add a (temporary) feature to log vsp ticket info from the staking page to confirm the status of purchased tickets. This will help with debugging ticket purchase issues especially pertaining to vsp fee payment.

Uses https://github.com/planetdecred/dcrlibwallet/pull/225.

### CODE REVIEW

File: ui/load/wallet.go [hunk 1]
```diff
@@ -40,6 +40,16 @@ func (wl *WalletLoad) SortedWalletList() []*dcrlibwallet.Wallet {
...
 
+// TicketPrice returns the current ticket price.
+func (wl *WalletLoad) TicketPrice() int64 {
+	pr, err := wl.MultiWallet.WalletsIterator().Next().TicketPrice()
+	if err != nil {
+		log.Error(err)
+		return 0
+	}
+	return pr.TicketPrice
+}
```
Comment (Sirmorrison) [inline ui/load/wallet.go]: This is similar to what we currently have `Multiwallet.TicketPrice()`. I dont think we need it.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
I get this error when running the app.
```
HEAD is now at 7b24e77 log vsp ticket info when possible
sirmorrison@sirmorrisons-MacBook-Pro godcr-gio % ./godcr --network==testnet
Config file (/Users/sirmorrison/Library/Application Support/Godcr/godcr.conf) does not exist. Using defaults.
2022-02-02 07:48:25.954 [ERR] UI: init multiwallet error: invalid net type

```

File: ui/page/staking/overview.go [hunk 2]
```diff
@@ -593,14 +602,7 @@ func (pg *Page) startTicketBuyerPasswordModal() {
...
 			go func() {
-				vsp, err := pg.WL.MultiWallet.NewVSPClient(tbConfig.VspHost, pg.ticketBuyerWallet.ID, uint32(tbConfig.PurchaseAccount))
-				if err != nil {
-					pg.Toast.NotifyError(err.Error())
-					pm.SetLoading(false)
-					return
-				}
-
-				err = vsp.StartTicketBuyer(tbConfig.BalanceToMaintain, []byte(password))
+				err := pg.ticketBuyerWallet.StartTicketBuyer(tbConfig.VspHost, nil, tbConfig.PurchaseAccount, tbConfig.BalanceToMaintain, []byte(password))
```
Comment (itswisdomagain) [inline ui/page/staking/overview.go]: I'm going to modify this to only take the password. The other values can be gotten from wallet config.

File: ui/page/staking/purchase_modal.go [hunk 3]
```diff
@@ -350,8 +351,12 @@ func (tp *stakingModal) Handle() {
 	}
 
 	// reselect vsp if there's a delay in fetching the VSP List
-	if len(tp.WL.MultiWallet.VspList) > 0 && tp.WL.MultiWallet.GetRememberVSP() != "" {
-		tp.vspSelector.selectVSP(tp.WL.MultiWallet.GetRememberVSP())
+	lastUsedVSP := tp.WL.MultiWallet.LastUsedVSP()
+	if len(tp.WL.MultiWallet.KnownVSPs()) > 0 && lastUsedVSP != "" {
+		// TODO: Handle is called anytime there's a UI change. Won't this
+		// cause the selection to always change? Although, technically no
+		// because the last used vsp is always updated.
```
Comment (Sirmorrison) [inline ui/page/staking/purchase_modal.go]: If this is not affected and will not be modified, we really don't need this comment.

File: ui/page/staking/vsp_selector.go [hunk 4]
```diff
@@ -44,7 +45,8 @@ func (v *vspSelector) Changed() bool {
 }
 
 func (v *vspSelector) selectVSP(vspHost string) {
-	for _, vsp := range v.WL.MultiWallet.VspList {
+	vsps := v.WL.MultiWallet.KnownVSPs()
+	for _, vsp := range vsps {
```
Comment (Sirmorrison) [inline ui/page/staking/vsp_selector.go]:
```suggestion
	for _, vsp := range v.WL.MultiWallet.KnownVSPs() {
```

I dont think there is need assigning this to a variable when it is only used once.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Comment by Sirmorrison** [comment]
PR functionality looks good just the few changes.

**Review by Sirmorrison — APPROVED** [review]

**Pull request merged**